src/Raven.Client/Documents/Queries/VectorFieldFactory.cs

```diff
@@ -0,0 +1,227 @@
+    {
+        FieldName = fieldName;
+        SourceQuantizationType = storedEmbeddingQuantization;
+        DestinationQuantizationType = SourceQuantizationType;
+        VectorIndexingStrategy = vectorIndexingStrategy;
+        
+        return this;
+    }
+
+    IVectorEmbeddingField IVectorFieldFactory<T>.WithEmbedding(Expression<Func<T, object>> propertySelector, EmbeddingType storedEmbeddingQuantization, VectorIndexingStrategy vectorIndexingStrategy)
+    {
+        FieldName = propertySelector.ToPropertyPath(DocumentConventions.Default);
+        SourceQuantizationType = storedEmbeddingQuantization;
+        DestinationQuantizationType = SourceQuantizationType;
+        VectorIndexingStrategy = vectorIndexingStrategy;
+        
+        return this;
+    }
+
+    IVectorEmbeddingField IVectorFieldFactory<T>.WithBase64(string fieldName, EmbeddingType storedEmbeddingQuantization, VectorIndexingStrategy vectorIndexingStrategy)
+    {
+        FieldName = fieldName;
+        SourceQuantizationType = storedEmbeddingQuantization;
+        DestinationQuantizationType = SourceQuantizationType;
+        IsBase64Encoded = true;
+        VectorIndexingStrategy = vectorIndexingStrategy;
+
+        return this;
+    }
+
+    IVectorEmbeddingField IVectorFieldFactory<T>.WithBase64(Expression<Func<T, object>> propertySelector, EmbeddingType storedEmbeddingQuantization, VectorIndexingStrategy vectorIndexingStrategy)
+    {
+        FieldName = propertySelector.ToPropertyPath(DocumentConventions.Default);
+        SourceQuantizationType = storedEmbeddingQuantization;
+        DestinationQuantizationType = SourceQuantizationType;
+        IsBase64Encoded = true;
+        VectorIndexingStrategy = vectorIndexingStrategy;
+
+        return this;
+    }
+    
+    IVectorField IVectorFieldFactory<T>.WithField(string fieldName)
+    {
+        FieldName = fieldName;
+        _byFieldMethodUsed = true;        
+        return this;
+    }
+
+    IVectorField IVectorFieldFactory<T>.WithField(Expression<Func<T, object>> propertySelector)
+    {
+        FieldName = propertySelector.ToPropertyPath(DocumentConventions.Default);
+        _byFieldMethodUsed = true;
+        return this;
+    }
+
+    IVectorEmbeddingField IVectorEmbeddingField.TargetQuantization(EmbeddingType targetEmbeddingQuantization)
+    {
+        PortableExceptions.ThrowIf<InvalidOperationException>(_byFieldMethodUsed, $"Cannot use method {nameof(IVectorEmbeddingField.TargetQuantization)} with {nameof(IVectorFieldFactory<T>.WithField)} since quantization is already done by the index.");
+        
+        DestinationQuantizationType = targetEmbeddingQuantization;
+        
+        if (SourceQuantizationType is EmbeddingType.Int8 or EmbeddingType.Binary && DestinationQuantizationType != SourceQuantizationType)
+            throw new InvalidOperationException("Cannot quantize already quantized embeddings.");
```

Comment:
you need to write those values in the ex, if we will not, user will not know what is going on exactly